pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarMetadataEventSynchronizer.java

```diff
@@ -181,19 +238,64 @@ private void startConsumer() {
     }
 
     public boolean isStarted() {
-        return started;
+        return this.state != State.Started;
+    }
+
+    public boolean isProducerStarted() {
+        return this.state.ordinal() > State.Starting_Producer.ordinal()
+                && this.state.ordinal() < State.Closing.ordinal();
+    }
+
+    public boolean isClosed() {
+        return this.state == State.Closing || this.state == State.Closed;
     }
 
     @Override
-    public void close() {
-        started = false;
-        if (producer != null) {
-            producer.closeAsync();
-            producer = null;
+    public synchronized CompletableFuture<Void> closeAsync() {
+        while (true) {
+            if (isClosed()) {
```

Comment:
```suggestion
            if (isClosingOrClosed()) {
```

Reply:
renamed the method `isClosed` to `isClosingOrClosed`

Reply:
Since the implementation of the method `isClosed` and `isClosingOrClosed` are the same, we just renamed it, so the test does not need to change